### DISCUSSION TITLE
Ask for “Official” Discord, Reddit Communities

### DISCUSSION THREAD

**Discussion by macoughl** [modelcontextprotocol/modelcontextprotocol#1094] [General]
### Pre-submission Checklist

- [X] I have verified that this discussion would not be more appropriate as an issue in a specific repository
- [X] I have searched existing discussions to avoid duplicates

### Discussion Topic

Hi Team,

I wanted to bring a discussion about governance of official MCP-related communities to the table. Specifically Discord and Reddit, but this may have other applications as well.

Ideally there would be  an “official” MCP community on each of these major platforms.

The current links on the [/servers README](https://github.com/modelcontextprotocol/servers) point to a Discord server owned by [glama.ai](http://glama.ai/), and the Reddit community [r/mcp](https://www.reddit.com/r/mcp/) (which is also controlled solely by the owner of glama.ai). I have nothing against @punkpeye specifically (though we did have a short back-and-forth I’ll ref below as an example of why we need, at minimum, distributed leadership here) - I would want to raise this issue regardless of which single (commercially-motivated) individual was at the helm of these communities. It looks like @jspahrsummers may have had similar concerns here: https://github.com/modelcontextprotocol/servers/pull/252

(Disclosure: I’m one of the three folks behind [Pulse](https://www.pulsemcp.com/) but we have zero commercial motivation and don't intend to)

I’ve observed at least two recent instances where I felt like these communities were not being managed in an unbiased manner:

- [Dedicated post warning against usage of Smithery](https://www.reddit.com/r/mcp/comments/1hg9u8f/be_careful_with_using_smithery/) - while the post in isolation is not technically incorrect, Smithery happens to be a direct alternative to glama.ai’s MCP directory. Which makes this post (one of the first-ever negative awareness posts in the subreddit) feel at least partly competitively motivated.

- Launch of Pulse [on Reddit](https://www.reddit.com/r/mcp/comments/1hhar9m/pulse_a_community_resource_to_browse_servers/), [ensuing Discord accusation of spam](https://discord.com/channels/1312302100125843476/1312302100125843479/1319039529809739816) - after I posted about Pulse (a curation of resources, among them an MCP directory) on Reddit, a public Discord message was made accusing us of “spam” and a threat to take it down. This is the type of thing that works against adoption of a protocol like this, not in favor of it.

To avoid similar situations in the future, I’m raising this issue for discussion so we can nip this in the bud and come up with a good structure now so we’re not having much harder conversations six months down the road.

I started this thread as a discussion around pain points like the examples above and how we want to better foster community here. 

To get the ball rolling, here are two proposals/ideas:

### Anthropic designates an employee to manage the communities
This is probably the simplest solution – as the creators of the protocol, Anthropic can be trusted to push the best interests of the protocol, and thus the community as a whole, forward in an unbiased manner (though are hope is that other LLMs are brought into the fold in the near future with equal “managing”/”ownership” capabilities to further adoption).

Of course, this is both a resource-ask of Anthropic, and introduces the Anthropic brand into the community in a way you may not want. So there may be dealbreaking reasons to not pursue this.

### Community/Anthropic nominates at least three separate third party individuals to co-moderate the communities
If the initial creators can't/don't want to moderate the communities, then the next best thing is to decentralize the responsibility across non-affiliated individuals. In the event of any disagreements, the three of them can work it out. There is little risk of any one individual allowing bias or preference for their own business or interests unduly influencing the community, because the other two would be there to counterbalance. Not nominating myself.

What do other folks think here?

**Comment by orliesaurus** [comment]
Hi everyone, I'm the guy behind [mcp-toolhouse-server](https://github.com/toolhouse-community/mcp-server-toolhouse) I am also active on Discord and X on the topic of MCP. First of all thanks for working on MCP, everyone, it's truly nice to see everyone's involvment.

I've been mostly asking questions about MCP lately as I've worked on our own internal draft for a tool-use protocol (I called it tool-use-spec internally)...but now it feels like it's better to help out MCP grow rather than put out another spec!

1/ I agree that unbiased management is crucial for community trust and protocol adoption, with the goal of distributed leadership to mitigate individual biases. I am +1 the idea of having a dedicated Anthropic lead until the project becomes more widespread and matures - a good indicator is when other large entities start backing it up - don't get me wrong, it's a great project but it's still in its infancy in terms of adoption.

2/ We need a direct alignment with the initial creators - at least at the start - for consistency in vision and objectives. I truly believe if Anthropic (or others) wanted to let this project go up into chaotic flames, they would ignore threads like this. Hopefully they won't. The community cannot be run by anyone else at the start as it might cause this kind of situation as OP has listed

3/ For MCP to thrive - like most successful community - we need to develop and document guidelines for moderation, decision-making, and conflict resolution. Include protocols (no pun intended) for handling biased or competitive content. i.e. who decides what is an "official server" and what isn't?

4/ Regarding censorship/deleting content: allow community members to voice their opinions - use the usual downvoting system to get rid of low quality effort content where possible. Don't allow single individuals to take content down - please!

I believe this would lead to transparency in decision-making processes which is essential for community building. This might require some folks to step up from Anthropic: it would be a good idea to start an outline of roles, responsibilities, and processes to inspire a trustworthy community. Not everything has to be rolled out at once!


P.S. Some community members may prefer the existing structure, that's cool this is just my opinion of course:
I voiced more questions here: https://news.ycombinator.com/item?id=42444484

**Comment by QuantGeekDev** [comment]
Hey! Thanks for the suggestion. We do have a community subreddit and a discord. It's called r/modelcontextprotocol and the Discord is there too. The r/mcp server is a clone created by @Punkpeye a few weeks after we made the modelcontextprotocol sub. He is solely interested in advertising his startup, Glama. He did this maliciously, being fully aware of an existing community, he purchased the /mcp sub which was a minecraft subreddit at the time. He then went on to fracture the community for the sake of profit.
Join the disucssion along with 1600 other people in r/modelcontextprotocol - we don't have any sponsors, no bs, just nerds talking mcp :)

**Reply by macoughl** [reply]
I'm embarrassed that I hadn't found that until now. Thank you so much for the direction - joined!

**Comment by QuantGeekDev** [comment]
The real community is on r/modelcontextprotocol

**Comment by QuantGeekDev** [comment]
We are also gearing up for an mcp hackathon next month - make sure you don't miss out :) it will be super fun

**Comment by QuantGeekDev** [comment]
If anybody from anthropic wants to reach out, my email: alex@andru.codes

We can discuss handling over the community over email

**Comment by QuantGeekDev** [comment]
@wong2 That is @punkpeye 's Modus Operandi.  He clones everyone's project as fast as he can, slaps his branding on it, and republishes it. Takes credit, and then uses a bot army to drown out the original authors. It is not nice, especially considering how young our community is, and how he is well aware of the damage he is doing by fracturing the community and creating all this spam. All of this for the sake of profit, abusing the free work of others. It has put me off from further contributions, at least while his name and spam is listed on Anthropic's official repo which is the biggest slap in the face

**Comment by jspahrsummers** [comment]
We're discussing within our team how we would like to manage/moderate/participate in Reddit, Discord, and other communities going forward. These would add significant support burden on our side, so we'd only want to invest after careful consideration.

Until that question is resolved, @modelcontextprotocol on GitHub remains our only official forum, and all other communities should be considered unofficial and unaffiliated. To ensure folks can find unofficial discussion forums as well as official ones, we'll review PRs that add links to communities into the README of the servers repo.

As this is one of our official forums, **please remember to abide by our [code of conduct](https://github.com/modelcontextprotocol/.github/blob/main/CODE_OF_CONDUCT.md)**, and refrain from personal attacks and the like. We will be moderating comments and users as necessary to ensure a positive environment for all.

**Comment by localden** [comment]
We've limited Discussions in this repository to meeting notes and are closing this thread. To continue the conversation, pick a channel from [Contributor Communication](https://modelcontextprotocol.io/community/communication). Thanks for posting.
